### ISSUE TITLE
Does not work on vanilla server

### ISSUE THREAD

**Issue by endumiuz** [theorbtwo/RoughlyEnoughResources#11]
I think Just Enough Resources works on vanilla servers, would It be possible to make this mod work on vanilla servers as well?

Edit: I assume it doesn't work because of how it gets data for the graphs (I haven't checked the source code so I could be wrong), but would it be possible to have a fallback for vanilla servers (or servers without the mod installed) that basically uses pregenerated data from a single player world?

**Comment by theorbtwo** [comment]
You are correct about why it doesn't work, but I would really prefer not to implement such a feature, since it is inherently fragile (each new version of vanilla will change world generation a bit, making any data files I store incorrect), and would require a fair bit of effort to implement.  Additionally, the information it seeks to convey isn't hard to find on the internet (though I will agree that it's a bit immersion-breaking to keep alt-tabbing out to a web browser).
If somebody else (you, for example) implements it, I will consider merging it, depending on how much it impacts the code, and how easy it is to regenerate the data files for a new version of vanilla.

**Comment by TelepathicGrunt** [comment]
the pregenerated data would be impossible to verify if correct as a vanilla server could use datapacks to change worldgen. It's not worth the effort to generate an entirely new world on the client (significantly increasing load times) to get supposedly untouched vanilla data. And servers that are modded without RER on can have mods that also change worldgen (and these mods can be entirely serverside) which RER on the client would not know about. Further complicating the matter is the client does not know what seed is used for the world too which would make it even harder to try and reproduce the same world on the client.

TLDR: Make the server be modded with RER on if you want the data. Otherwise, it's a technical nightmare to do it entirely clientsided and could possibly annoy some stricter server owners who don't like clients having mods that could expose the server's world seed inadvertently.

**Comment by theorbtwo** [comment]
Thanks, @TelepathicGrunt, you've explained the issue far more lucidly then I have managed elsewhere.  I'm going to keep this bug open but mark it WONTFIX, because RER is working as designed, but this is a common complaint.  Now I have somewhere to refer all the other bugs to, and hopefully a few folks will read this before reporting this as a bug.

**Comment by Evolution0** [comment]
> Thanks, @TelepathicGrunt, you've explained the issue far more lucidly then I have managed elsewhere. I'm going to keep this bug open but mark it WONTFIX, because RER is working as designed, but this is a common complaint. Now I have somewhere to refer all the other bugs to, and hopefully a few folks will read this before reporting this as a bug.

Good thing you did because I was scratching my head wondering why I couldn't get ore distribution information to show up on my modded client but vanilla server ^_^

Curious if some kind of helper plugin for non-mod supported servers like PaperMC can pass the required information to RER?

**Comment by furuochen-dev** [comment]
Can it be there's a option that make the mod use vanilla loot table?
